frontend/src/app/offers/create-offer/create-offer.component.ts

```diff
@@ -0,0 +1,192 @@
+      content: base64,
+      filename: 'image.jpg'
+    }))
+    .subscribe((response: BaseOffer) => {
+      this.offer = response as AppOffer;
+      this.inEditMode = true;
+      this.offer = this.removeTime();
+      this.form.patchValue(this.offer);
+    });
+  }
+
+// Temporary fix to make editing work properly, to delete after we fix datetime issue
+  removeTime(){
+    this.offer.startedAt = this.offer.startedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    this.offer.finishedAt = this.offer.finishedAt.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    if(this.offer.recruitmentEndDate){
+      this.offer.recruitmentEndDate = this.offer.recruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if(this.offer.recruitmentStartDate){
+      this.offer.recruitmentStartDate = this.offer.recruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    };
+    if(this.offer.reserveRecruitmentEndDate){
+      this.offer.reserveRecruitmentEndDate = this.offer.reserveRecruitmentEndDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    if(this.offer.reserveRecruitmentStartDate){
+      this.offer.reserveRecruitmentStartDate = this.offer.reserveRecruitmentStartDate.replace(/T[0-9][0-9]:[0-9][0-9]:[0-9][0-9]Z$/, '');
+    }
+    return this.offer;
+  }
+
+  // TODO - delete when we decide what date format we want to have
+  addTime(){
+    this.form.value.startedAt = this.form.value.startedAt + 'T00:00:00Z';
+    this.form.value.finishedAt = this.form.value.finishedAt + 'T00:00:00Z';
+
+    if (this.form.value.recruitmentEndDate) {
+      this.form.value.recruitmentEndDate = this.form.value.recruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.recruitmentStartDate) {
+      this.form.value.recruitmentStartDate = this.form.value.recruitmentStartDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentEndDate) {
+      this.form.value.reserveRecruitmentEndDate = this.form.value.reserveRecruitmentEndDate + 'T00:00:00Z';
+    }
+    if (this.form.value.reserveRecruitmentStartDate) {
+      this.form.value.reserveRecruitmentStartDate = this.form.value.reserveRecruitmentStartDate + 'T00:00:00Z';
+    }
+  }
+
+  ngOnDestroy() {
+    if (this.userSubscription) {
+      this.userSubscription.unsubscribe();
+    }
+  }
+
+  toDataUrl(url, callback) {
+    const xhr = new XMLHttpRequest();
+    xhr.onload = function() {
+        const reader = new FileReader();
+        reader.onloadend = function() {
+            callback(reader.result);
+        }
+        reader.readAsDataURL(xhr.response);
+    };
+    xhr.open('GET', url);
+    xhr.responseType = 'blob';
+    xhr.send();
+}
+
+  onSubmit(offer: AppOffer) {
+    if(this.form.valid){
+      this.addTime()
+    }
+    if(this.offer.image){
+      this.form.value.image = this.offer.image;
+      this.form.value.image.content = this.form.value.image.content.replace(/^data:image\/[a-z]+;base64,/, "");
```

Comment:
Your regExp seems to work pretty good :) However, in my opinion, it is a bit too complicated for this particular task . We just care to replace everything before comma (comma included) with the empty string, so the code `.replace(/*.,/, "");` seems to be enough.

As a side note, I usually use this website to test regExps: [link](https://regex101.com/)